flexget/plugins/api_tvmaze.py

```diff
@@ -0,0 +1,336 @@
+            pytvmaze_show = get_show(**prepared_params)
+        except ShowNotFound:
+            log.debug('could not find series {0} in pytvmaze'.format(title))
+            return
+
+        # See if series already exist in cache
+        series = session.query(TVMazeSeries).filter(TVMazeSeries.maze_id == pytvmaze_show.maze_id).first()
+        if series:
+            log.debug('found expired series {0}, refreshing data.'.format(series.name))
+            series.update(pytvmaze_show, session)
+            series = populate_episodes(series_object=series, show_data=pytvmaze_show, session=session)
+            session.flush()
+        else:
+            log.debug('creating new series {0} in tvmaze_series db'.format(title))
+            series = TVMazeSeries(pytvmaze_show, session)
+            series = populate_episodes(series_object=series, show_data=pytvmaze_show, session=session)
+            session.add(series)
+        # If there's a mismatch between actual series name and requested title,
+        # add it to lookup table for future lookups
+        if series and title.lower() != series.name.lower():
+            log.debug('mismatch between series title and search title. saving in lookup table')
+            session.add(TVMazeLookup(search_name=title, series=series))
+        return series
+
+    @staticmethod
+    @with_session
+    def episode_lookup(session=None, force_cache=False, **lookup_params):
+        series_name = lookup_params.get('series_name')
+        season_number = lookup_params.get('series_season')
+        episode_number = lookup_params.get('series_episode')
+        if not all([season_number, episode_number, series_name]):
+            raise LookupError('Not enough parameter to lookup episode')
+        series = APITVMaze.series_lookup(session=session, force_cache=force_cache, **lookup_params)
+        if not series:
+            raise LookupError('Could not find series with the following parameters: {0}'.format(**lookup_params))
+        for episode in series.episodes:
+            if episode.season_number == season_number and episode.number == episode_number:
+                return episode
+        return
+
+    @staticmethod
+    @with_session
+    def episode_airdate(session=None, force_cache=False, **lookup_params):
+        series_name = lookup_params.get('series_name')
+        season_number = lookup_params.get('series_season')
+        episode_number = lookup_params.get('series_episode')
+        if not all([season_number, episode_number, series_name]):
+            raise LookupError('Not enough parameter to lookup episode')
+        episode = APITVMaze.episode_lookup(session=session, force_cache=force_cache, **lookup_params)
+        if not episode:
+            raise LookupError(
+                'Could not find episode {0}, season {1} for show{2}'.format(episode_number, season_number, series_name))
+        return episode.airdate
```

Comment:
Is this really needed? Can't you just do an episode lookup and check the airdate?